infrastructure/tooling/src/publish/index.js

```diff
@@ -37,21 +37,22 @@ class Publish {
       }
     }
 
-    if (this.cmdOptions.push) {
-      [
-        'GH_TOKEN',
-        'NPM_TOKEN',
-        'PYPI_USERNAME',
-        'PYPI_PASSWORD',
-        'DOCKER_USERNAME',
-        'DOCKER_PASSWORD',
-      ].forEach(e => {
-        if (!process.env[e]) {
-          throw new Error(`$${e} is required (unless --no-push)`);
-        }
-      });
+    const expectedVars = [];
+    expectedVars.push('GH_TOKEN');
+    if (!this.cmdOptions.staging) {
+      expectedVars.push('NPM_TOKEN');
+      expectedVars.push('PYPI_USERNAME');
+      expectedVars.push('PYPI_PASSWORD');
```

Comment:
iirc there are ways to do staging releases for npm/pypi as well? We don't have to do it now but maybe in the space future we can stage releases to all of these or something.

Reply:
For pypi, yes -- there's a `test.pypi.org`.  Maybe NPM, too?  But honestly pushing to docker hub would probably be the most useful..